core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -130,14 +131,16 @@ public int getAdvertisedPort() {
      * Gets the host name that we advertise protocol clients to connect to.
      * @since 2.198
      */
-    public String getAdvertisedHost() {
+    public String getAdvertisedHost() throws URISyntaxException {
         if (CLI_HOST_NAME != null) {
           return CLI_HOST_NAME;
         }
         try {
-            return new URL(Jenkins.get().getRootUrl()).getHost();
+            return new URI(Jenkins.get().getRootUrl()).toURL().getHost();
         } catch (MalformedURLException e) {
             throw new IllegalStateException("Could not get TcpSlaveAgentListener host name", e);
+        } catch (URISyntaxException e) {
```

Comment:
this should be handled the same as if the URL provided was garbage - see the line above,  you can even use multi-catch here

Reply:
Done. See below
```
} catch (MalformedURLException | URISyntaxException e) {
            throw new IllegalStateException("Could not get TcpSlaveAgentListener host name", e);
        }
```